src/Raven.Server/ServerWide/SubscriptionsClusterStorage.cs

```diff
@@ -25,6 +27,18 @@ public SubscriptionState ReadSubscriptionStateByName(ClusterOperationContext con
             throw new SubscriptionDoesNotExistException($"Subscription with name '{name}' was not found in server store");
 
         var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+        
+        if (subscriptionState.ArchivedDataProcessingBehavior.HasValue)
+            return subscriptionState;
+        
+        // persisted state from v5.x
+        if(Enum.TryParse(_cluster.ReadDatabase(context, databaseName).Settings[RavenConfiguration.GetKey(x => x.Subscriptions.ArchivedDataProcessingBehavior)], false,
+               out ArchivedDataProcessingBehavior archivedDataProcessingBehavior) == false)
+        {
+            throw new InvalidOperationException(
+                $"Failed to fetch {nameof(RavenConfiguration.Subscriptions.ArchivedDataProcessingBehavior)} from subscriptions configuration in database settings.");
+        }
+        subscriptionState.ArchivedDataProcessingBehavior = archivedDataProcessingBehavior;
         return subscriptionState;
```

Comment:
not sure if this is a good place, should be done in a place that is aware of database/sharded-context, probably in impl of `AbstractSubscriptionStorage<AbstractSubscriptionConnectionsState>)`

Reply:
moved it out as mentioned